### PR TITLE
Add boundary value tests for partial decoding

### PR DESCRIPTION
**Pull request by keiji**
Added comprehensive boundary value tests for partial decoding in `test/test_wrapper.c`. This includes checking for `ERR_REGION_OUT_OF_BOUNDS` when coordinates are outside the image dimensions or invalid (e.g. negative/underflow), and ensuring valid boundary coordinates are accepted by the logic. Verified with coverage analysis.

---
*PR created automatically by Jules for task [15188970708946868904](https://jules.google.com/task/15188970708946868904) started by @keiji*

### CODE REVIEW

File: test/test_wrapper.c [hunk 1]
```diff
@@ -589,6 +589,106 @@ void test_ratio_decode() {
...
+    assert(last_error == ERR_REGION_OUT_OF_BOUNDS);
+
+    // 16. Right Limit (x1=100), Bottom Exceed (y1 > 100)
+    result = decodeToBmp(dummy_data, 20, 0, 1000, COLOR_FORMAT_ARGB8888, 90, 90, 100, 101);
+    assert(result == NULL);
+    assert(last_error == ERR_REGION_OUT_OF_BOUNDS);
+
+    printf("Detailed Boundaries Passed.\n");
+    stub_should_header_succeed = 0;
+}
```
[the remark this answers is not in the document]
Comment (keiji) [reply test/test_wrapper.c:661]: medium

This new test function `test_detailed_boundaries` contains a significant amount of duplicated code. Each of the 20 test cases repeats the same pattern of calling `decodeToBmp` and asserting the results. This repetition makes the test function long, hard to read, and difficult to maintain.

Consider refactoring this by creating a helper function that encapsulates the repeated logic. This would make the test cases much more concise and declarative.

For example, you could have a helper like this:

```c
void check_decode_boundary(const char* test_name, uint8_t* data, uint32_t x0, uint32_t y0, uint32_t x1, uint32_t y1, int expected_error) {
    uint8_t* result = decodeToBmp(data, 20, 0, 1000, COLOR_FORMAT_ARGB8888, x0, y0, x1, y1);
    assert(result == NULL);
    if (last_error != expected_error) {
        printf("Test '%s' failed: expected error %d, got %d\n", test_name, expected_error, last_error);
        assert(last_error == expected_error);
    }
}
```

And then call it for each case:

```c
check_decode_boundary("1. Left-Up Limit: 0, 0, 10, 10 (Valid)", dummy_data, 0, 0, 10, 10, ERR_DECODE);
check_decode_boundary("2. Left-Up Exceed", dummy_data, UINT32_MAX, UINT32_MAX, 10, 10, ERR_REGION_OUT_OF_BOUNDS);
// ...etc
```

File: test/test_wrapper.c [hunk 2]
```diff
@@ -589,6 +589,106 @@ void test_ratio_decode() {
...
+    assert(last_error == ERR_REGION_OUT_OF_BOUNDS);
+
+    // 3. Right-Up Limit: 90, 0, 100, 10 (Valid)
+    result = decodeToBmp(dummy_data, 20, 0, 1000, COLOR_FORMAT_ARGB8888, 90, 0, 100, 10);
+    assert(result == NULL);
+    assert(last_error == ERR_DECODE);
+
+    // 4. Right-Up Exceed: (Right Exceeds)
+    // x1 > 100
+    result = decodeToBmp(dummy_data, 20, 0, 1000, COLOR_FORMAT_ARGB8888, 90, UINT32_MAX, 101, 10);
```
[the remark this answers is not in the document]
Comment (keiji) [reply test/test_wrapper.c]: The comments for this test case are misleading. The title Right-Up Exceed: (Right Exceeds) and the sub-comment // x1 > 100 suggest that only the right boundary is being tested. However, the parameters to decodeToBmp also include y0=UINT32_MAX, which tests an 'Up Exceeds' condition. To improve clarity and consistency with other corner exceed tests (like test 2 and 8), the comments should accurately reflect that both right and up boundaries are being exceeded.

File: test/test_wrapper.c [hunk 3]
```diff
@@ -589,6 +589,106 @@ void test_ratio_decode() {
...
+    assert(last_error == ERR_REGION_OUT_OF_BOUNDS);
+
+    // 5. Left-Down Limit: 0, 90, 10, 100 (Valid)
+    result = decodeToBmp(dummy_data, 20, 0, 1000, COLOR_FORMAT_ARGB8888, 0, 90, 10, 100);
+    assert(result == NULL);
+    assert(last_error == ERR_DECODE);
+
+    // 6. Left-Down Exceed: (Down Exceeds)
+    // y1 > 100
+    result = decodeToBmp(dummy_data, 20, 0, 1000, COLOR_FORMAT_ARGB8888, UINT32_MAX, 90, 10, 101);
```
[the remark this answers is not in the document]
Comment (keiji) [reply test/test_wrapper.c]: Similar to a previous test case, the comments here are misleading. The title Left-Down Exceed: (Down Exceeds) and sub-comment // y1 > 100 imply only the lower boundary is tested. However, the code uses x0=UINT32_MAX, which also tests a 'Left Exceeds' condition. For clarity and consistency, the comments should be updated to state that both left and down boundaries are being tested.
